ksql-streams/src/main/java/io/confluent/ksql/execution/streams/GroupByParamsFactory.java

```diff
@@ -82,16 +93,36 @@ private static LogicalSchema buildSchema(
   private static Object processColumn(
       final int index,
       final ExpressionMetadata exp,
-      final GenericRow row
+      final GenericRow row,
+      final ProcessingLogger logger
   ) {
     try {
       final Object result = exp.evaluate(row);
       if (result == null) {
+        logger.error(
+            EngineProcessingLogMessageFactory.recordProcessingError(
+               String.format(
+                   "Group-by column with index %d resolved to null."
+                       + " The source row will be excluded from the table.",
+                   index),
+                row
+            )
+        );
```

Comment:
Do we want to remove the `LOG.error` line below now that we have this?  Seems unnecessary to log the error twice. 

Other uses of the processing logger don't double log.